app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -235,6 +238,12 @@ class Settings private constructor(context: Context) {
             .apply()
     }
 
+    fun setHasAuthenticationProblem(hasProblem: Boolean) {
```

Comment:
Instead of keeping around extra state, I'd prefer it if you used `accountManager.accountNeedsReauth` if possible. This makes things conceptually simpler, since you won't need to worry about an extra flag going stale and out-of-sync with the actual account state.